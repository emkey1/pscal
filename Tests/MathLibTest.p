program MathLibTest;

uses
  MathLib;

const
  TOL = 0.0001;

var
  r: real;
  i: integer;

procedure TestMathLib;
begin
  writeln;
  writeln('--- Testing MathLib ---');

  r := ArcTan(0.5);
  write('START: ArcTan(0.5): ');
  if abs(0.4636 - r) < TOL then
    writeln('PASS')
  else
    writeln('FAIL (expected: ', 0.4636:0:4, ', got: ', r:0:4, ')');

  r := ArcSin(0.5);
  write('START: ArcSin(0.5): ');
  if abs(0.5236 - r) < TOL then
    writeln('PASS')
  else
    writeln('FAIL (expected: ', 0.5236:0:4, ', got: ', r:0:4, ')');

  r := ArcCos(0.5);
  write('START: ArcCos(0.5): ');
  if abs(1.0472 - r) < TOL then
    writeln('PASS')
  else
    writeln('FAIL (expected: ', 1.0472:0:4, ', got: ', r:0:4, ')');

  r := Cotan(0.785398);
  write('START: Cotan(pi/4): ');
  if abs(1.0 - r) < TOL then
    writeln('PASS')
  else
    writeln('FAIL (expected: ', 1.0:0:4, ', got: ', r:0:4, ')');

  r := Power(2.0, 3.0);
  write('START: Power(2,3): ');
  if abs(8.0 - r) < TOL then
    writeln('PASS')
  else
    writeln('FAIL (expected: ', 8.0:0:4, ', got: ', r:0:4, ')');

  r := Log10(1000.0);
  write('START: Log10(1000): ');
  if abs(3.0 - r) < TOL then
    writeln('PASS')
  else
    writeln('FAIL (expected: ', 3.0:0:4, ', got: ', r:0:4, ')');

  r := Sinh(0.0);
  write('START: Sinh(0): ');
  if abs(0.0 - r) < TOL then
    writeln('PASS')
  else
    writeln('FAIL (expected: ', 0.0:0:4, ', got: ', r:0:4, ')');

  r := Cosh(0.0);
  write('START: Cosh(0): ');
  if abs(1.0 - r) < TOL then
    writeln('PASS')
  else
    writeln('FAIL (expected: ', 1.0:0:4, ', got: ', r:0:4, ')');

  r := Tanh(0.0);
  write('START: Tanh(0): ');
  if abs(0.0 - r) < TOL then
    writeln('PASS')
  else
    writeln('FAIL (expected: ', 0.0:0:4, ', got: ', r:0:4, ')');

  r := Max(2.0, 1.0);
  write('START: Max(2,1): ');
  if abs(2.0 - r) < TOL then
    writeln('PASS')
  else
    writeln('FAIL (expected: ', 2.0:0:4, ', got: ', r:0:4, ')');

  r := Min(2.0, 1.0);
  write('START: Min(2,1): ');
  if abs(1.0 - r) < TOL then
    writeln('PASS')
  else
    writeln('FAIL (expected: ', 1.0:0:4, ', got: ', r:0:4, ')');

  i := Floor(3.7);
  write('START: Floor(3.7): ');
  if 3 = i then
    writeln('PASS')
  else
    writeln('FAIL (expected: 3, got: ', i, ')');

  i := Ceil(3.1);
  write('START: Ceil(3.1): ');
  if 4 = i then
    writeln('PASS')
  else
    writeln('FAIL (expected: 4, got: ', i, ')');
end;

begin
  writeln('Running MathLib tests');
  TestMathLib;
  writeln;
  writeln('MathLib tests completed.');
end.
