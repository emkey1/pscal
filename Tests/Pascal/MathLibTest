program MathLibTest;

uses
  MathLib; { constants only; functions are tested via builtins }

const
  TOL = 0.0001;

procedure AssertEqualInt(expected, actual: integer; testName: string);
begin
  write('START: ', testName, ': ');
  if expected = actual then
    writeln('PASS')
  else
    writeln('FAIL (expected: ', expected, ', got: ', actual, ')');
end;

procedure AssertEqualReal(expected, actual: real; testName: string);
begin
  write('START: ', testName, ': ');
  if abs(expected - actual) < TOL then
    writeln('PASS')
  else
    writeln('FAIL (expected: ', expected:0:4, ', got: ', actual:0:4, ')');
end;

procedure TestMathLib;
begin
  writeln;
  writeln('--- Testing MathLib ---');
  AssertEqualReal(0.7854, arctan(1.0), 'arctan(1)');
  AssertEqualReal(0.5236, arcsin(0.5), 'arcsin(0.5)');
  AssertEqualReal(1.0472, arccos(0.5), 'arccos(0.5)');
  AssertEqualReal(1.0, cotan(0.785398), 'cotan(pi/4)');
  AssertEqualReal(8.0, power(2.0, 3.0), 'power(2,3)');
  AssertEqualReal(3.0, log10(1000.0), 'log10(1000)');
  AssertEqualReal(0.0, sinh(0.0), 'sinh(0)');
  AssertEqualReal(1.0, cosh(0.0), 'cosh(0)');
  AssertEqualReal(0.0, tanh(0.0), 'tanh(0)');
  AssertEqualReal(2.0, max(2.0, 1.0), 'max(2,1)');
  AssertEqualReal(1.0, min(2.0, 1.0), 'min(2,1)');
  AssertEqualInt(3, floor(3.7), 'floor(3.7)');
  AssertEqualInt(4, ceil(3.1), 'ceil(3.1)');
end;

begin
  writeln('Running MathLib tests');
  TestMathLib;
  writeln;
  writeln('MathLib tests completed.');
end.
