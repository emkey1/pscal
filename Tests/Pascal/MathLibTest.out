Running MathLib tests

--- Testing MathLib ---
START: arctan(1): PASS
START: arcsin(0.5): PASS
START: arccos(0.5): PASS
START: cotan(pi/4): PASS
START: power(2,3): PASS
START: log10(1000): PASS
START: sinh(0): PASS
START: cosh(0): PASS
START: tanh(0): PASS
START: max(2,1): PASS
START: min(2,1): PASS
START: floor(3.7): PASS
START: ceil(3.1): PASS

MathLib tests completed.
